src/lib/middleware/with-access-token.ts

```diff
@@ -14,53 +15,99 @@ export const extractShortToken = (token: string) => token.split("_")?.[1]
 export const extractLongTokenHash = (token: string) =>
   hashLongToken(extractLongToken(token) ?? "")
 
-export const withAccessToken: Middleware<
-  {
-    auth: {
-      auth_mode: "access_token"
+export const withAccessToken =
+  <RequiresWorkspaceId extends boolean>({
+    require_workspace_id,
```

Comment:
```suggestion
    is_workspace_id_required,
```

Reply:
can make this change, but was trying to align naming with current convention in seam-connect, is it worth drifting away?

Reply:
Sure, if some of this is copy/paste from seam connect, then it's ok to follow that. I can't really know the difference in review, so I just have to flag things that look like they break convention. You can apply best judgement.